src/layouts/Settings.jsx

```diff
@@ -0,0 +1,506 @@
+      const currentColor = value.slice(1);
+      this.setState((currState) => ({
+        ...currState,
+        colorPicker: {
+          colorPickerToggle: true,
+          currentColor,
+          elementId: id,
+          oldColors: { ...colors },
+        },
+      }));
+    }
+  }
+
+  disableColorPicker = (resetOldColors) => {
+    const { colorPicker: { colorPickerToggle, oldColors } } = this.state;
+    // if ColorPicker is active, disable it
+    if (colorPickerToggle) {
+      if (resetOldColors) {
+        this.setState((currState) => ({
+          ...currState,
+          colors: oldColors,
+        }));
+      }
+
+      this.setState((currState) => ({
+        ...currState,
+        colorPicker: {
+          colorPickerToggle: false,
+          currentColor: '',
+          elementId: '',
+          oldColors: {},
+        },
+      }));
+    }
+
+    // remove event listeners
+    document.removeEventListener('keydown', this.escFunction);
+    document.removeEventListener('click', this.clickFunction);
+  }
+
+  // onColorSelect sets value of appropriate color field
+  onColorSelect = (event, color) => {
+    // prevent event from reloading
+    // prevent parent form from being submitted
+    event.preventDefault();
+    event.stopPropagation();
+
+    // reflect color changes
+    this.realTimeColor(color);
+
+    // reset color picker
+    this.disableColorPicker(false);
+  }
+
+  realTimeColor = (color) => {
+    const { colorPicker: { elementId }, colors } = this.state;
+    // there is no hex property if the color submitted is the
+    // same as the original color
+    const hex = color.hex ? color.hex : `#${color}`;
+
+    // set state of color fields
+    if (elementId === 'primary-color' || elementId === 'secondary-color') {
+      this.setState((currState) => ({
+        ...currState,
+        colors: {
+          ...currState.colors,
+          [elementId]: hex,
+        },
+      }));
+    } else {
+      // set state of resource colors
+      const index = elementId.split('@')[elementId.split('@').length - 1];
+      const newMediaColors = _.cloneDeep(colors['media-colors']);
+      newMediaColors[index].color = hex;
+      this.setState((currState) => ({
+        ...currState,
+        colors: {
+          ...currState.colors,
+          'media-colors': newMediaColors,
+        },
+      }));
+    }
+  }
+
+  // a recursive function to retrieve all values in the nested error object
+  retrieveErrors = (errors) => {
+    if (errors.colors) {
+      errors.colors['media-colors'].forEach((curr) => {
+        // eslint-disable-next-line no-param-reassign
+        delete curr.title;
+      });
+    }
+    const flattenedErrors = Object.values(errors).reduce(this.retrieveErrorsCallback, []);
+    return _.some(flattenedErrors);
+  }
+
+  retrieveErrorsCallback = (acc, curr) => {
+    if (typeof curr === 'object') {
+      return acc.concat(this.retrieveErrors(curr));
+    }
+    return acc.concat(curr);
+  }
```

Comment:
Learnt that this is to check for any error values within a nested object. Don't think it's worth to do hard to read recursive functions just to check for error values within `media-colors`.

Reply:
Fixed with [50fd1f6c](https://github.com/isomerpages/isomercms-frontend/pull/99/commits/50fd1f6cce5e68395c71d8c2bc92340bee022e68)